src/operator/loss.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+
+from typing import List
+from src.operator.operator import Operator
+from src.operator.normalizer import Softmax
+from src.utils import check_shape
+
+
+class Loss(Operator):
+
+    def __init__(self, ishape: List[int]):
+        super().__init__(ishape, [])
+        self.preds = [None, None]
+
+    def register(self, pred: Operator):
+        raise NotImplemented(
+            'Regular register not supported for loss function, ' +
+            'use register_output and register_ground_truth')
+
+    def register_output(self, pred: Operator):
+        if self.preds[0] is not None:
+            raise Exception('Output for loss is already set')
+        check_shape(self.ishape, pred.oshape)
+        self.preds[0] = pred
+        pred.succs.append(self)
+
+    def register_ground_truth(self, pred: Operator):
+        if self.preds[1] is not None:
+            raise Exception('Ground truth for loss is already set')
+        check_shape(self.ishape, pred.oshape)
+        self.preds[1] = pred
+        pred.succs.append(self)
+
+    @property
+    def y(self):
```

Comment:
Suggest changing `y` to `y_pred` or `y_hat`, to make it explicit as the predicted output y

Reply:
Changed to `y_pred`